firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/CustomModelDownloadService.java

```diff
@@ -0,0 +1,247 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.ml.modeldownloader.internal;
+
+import android.os.Build.VERSION_CODES;
+import android.util.JsonReader;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import com.google.android.gms.common.util.VisibleForTesting;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.installations.FirebaseInstallationsApi;
+import com.google.firebase.installations.InstallationTokenResult;
+import com.google.firebase.ml.modeldownloader.CustomModel;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import javax.net.ssl.HttpsURLConnection;
+
+/**
+ * Calls the Download Service API and returns the information related to the current status of a
+ * custom model. If new model available returns 200 and new model details such as the download url;
+ * If same model is available returns 304 (not modified) Otherwise returns an error.
+ *
+ * @hide
+ */
+@RequiresApi(api = VERSION_CODES.KITKAT)
+final class CustomModelDownloadService {
+
+  private static final String TAG = "CustomModelDownloadSer";
+  private final ExecutorService executorService;
+
+  private static final int CONNECTION_TIME_OUT_MS = 2000; // 2 seconds.
+  private static final Charset UTF_8 = StandardCharsets.UTF_8;
+
+  private static final String ETAG_HEADER = "ETag";
+
+  @VisibleForTesting
+  static final String INSTALLATIONS_AUTH_TOKEN_HEADER = "X-Goog-Firebase-Installations-Auth";
+
+  @VisibleForTesting
+  static final String DOWNLOAD_MODEL_REGEX =
+      "https://firebaseml.googleapis.com/Model/v1beta2/projects/%s/models/%s:download";
```

Comment:
I think the URL needs to be: "https://firebaseml.googleapis.com/v1beta2/projects/%s/models/%s:download".